### PR TITLE
Fixed ai_agent flee.

### PR DESCRIPTION
**Pull request by cressidagp**
Hi there community, I give it a try to this and was able to make it to work.  Tested and working...

TODO: 

1. Use the packet stuff to send flee message
2. The walk speed its too high, they need to use some kind of slowWalk
3. Flee distance seems to be too short...
4. Most likely there should be a spell for handle this. Maybe this 31365

SQL file with all flee agents: https://pastebin.com/Qrgyi1Qe

### CODE REVIEW

**Comment by cressidagp** [comment]
Damn, i broke the linux build?

**Comment by dfighter1985** [comment]
> Damn, i broke the linux build?

Nope. I will just have to tune the pipeline to make it not want to push the docker images of every branch.

**Comment by dfighter1985** [comment]
Hi there!
I will review your patch soon (tm).

> SQL file with all flee agents: https://pastebin.com/Qrgyi1Qe

I have created a database project for Arcemu (forked ArcDB really), so content will go there from now on.
Here's where you can find it:
https://github.com/arcemu/db

File: src/world/Game/AI/AIInterface.cpp [hunk 1]
```diff
@@ -925,20 +925,20 @@ void AIInterface::_UpdateCombat(uint32 p_time)
...
-					string msg = "%s attempts to run away in fear!";
-					data << (uint8)CHAT_MSG_CHANNEL;
-					data << (uint32)LANG_UNIVERSAL;
-					data << (uint32)(strlen(TO< Creature* >(m_Unit)->GetCreatureInfo()->Name) + 1);
-					data << TO< Creature* >(m_Unit)->GetCreatureInfo()->Name;
-					data << (uint64)0;
-					data << (uint32)(msg.size() + 1);
-					data << msg;
-					data << uint8(0);
+					//WorldPacket data(SMSG_MESSAGECHAT, 100);
```
Comment (dfighter1985) [inline src/world/Game/AI/AIInterface.cpp:928]: Please remove code, not just comment it.

File: src/world/Game/AI/AIInterface.cpp [hunk 2]
```diff
@@ -925,20 +925,20 @@ void AIInterface::_UpdateCombat(uint32 p_time)
...
+					char msg[256] = "%s attempts to run away in fear!";
```
Comment (dfighter1985) [inline src/world/Game/AI/AIInterface.cpp:929]: Don't do this please!
Either make it a string literal ( const char* msg = "blahblah"; ), or a C++ string ( string msg = "blahblah"; ).

File: src/world/Game/Storage/ObjectStorage.cpp [hunk 3]
```diff
@@ -308,7 +308,8 @@ void ObjectMgr::LoadExtraCreatureProtoStuff()
 				{
 					cn->m_canFlee = true;
 					if(sp->floatMisc1)
-						cn->m_canFlee = (sp->floatMisc1 > 0.0f ? true : false);
```
Comment (dfighter1985) [inline src/world/Game/Storage/ObjectStorage.cpp:311]: Are we sure all creatures (that includes bosses, guards, etc too) will flee?

**Comment by cressidagp** [comment]
> Hi there! I will review your patch soon (tm).
> 
> > SQL file with all flee agents: https://pastebin.com/Qrgyi1Qe
> 
> I have created a database project for Arcemu (forked ArcDB really), so content will go there from now on. Here's where you can find it: https://github.com/arcemu/db

Hi there! i will contribute. Did some job while pandemy. Also worked at a little in Arcemu Fandom.

File: src/world/Game/Storage/ObjectStorage.cpp (hunk 3 above)
Comment (cressidagp) [reply src/world/Game/Storage/ObjectStorage.cpp:311]: uhm, i checked a couple of creatures wich dont have agent flee entry in the table and they wasnt fleeing... im gonna check again tomorrow.

Comment (cressidagp) [reply src/world/Game/Storage/ObjectStorage.cpp:311]: You want to have some check to prevent in case of someone add a flee agent for a boss or guard?

Comment (dfighter1985) [reply src/world/Game/Storage/ObjectStorage.cpp:311]: No I don't think it should be fool-proof, maybe I just misunderstood something. I will check again :)

Comment (dfighter1985) [reply src/world/Game/Storage/ObjectStorage.cpp:311]: OK, I get it now, I am just blind sometimes :D
I didn't notice that this code was only executing for the FLEE agent.
Sorry about that!

**Review by dfighter1985 — APPROVED** [review]
Ok, now I looks great!
Thanks!
I'll merge it soon.

**Comment by dfighter1985** [comment]
I will check the data too soon, and probably add to the Arcemu db project.

**Comment by cressidagp** [comment]
> Ok, now I looks great!
> Thanks!
> I'll merge it soon.
> 

I'm happy to have my little grain of sand.

**Pull request merged**